packages/kit/src/runtime/client/client.js

```diff
@@ -1857,15 +1857,17 @@ async function load_data(url, invalid) {
 
 	const res = await native_fetch(data_url.href);
 
+	if (res.headers.get('content-type')?.includes('text/html')) {
+		await native_navigation(url);
```

Comment:
I feel that's a better and more generic solution to our problem 👍 We should add a comment though explaining why this could happen.